### PR TITLE
rename TLSConfig to GatewayTLSConfig

### PR DESCRIPTION
**Pull request by hbagdi**

### CODE REVIEW

File: apis/v1alpha1/tlsconfig_types.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ const (
 // - gcp: https://cloud.google.com/load-balancing/docs/use-ssl-policies#creating_an_ssl_policy_with_a_custom_profile
 // - aws: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#describe-ssl-policies
 // - azure: https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-bindings#enforce-tls-1112
-type TLSConfig struct {
+type GatewayTLSConfig struct {
```
Comment (robscott) [inline apis/v1alpha1/tlsconfig_types.go:44]: At this point should we just move this into gateway_types.go?

Comment (hbagdi) [reply apis/v1alpha1/tlsconfig_types.go:44]: We can. It will conflict with #298

**Comment by bowei** [comment]
move to gateway_types.go (follow up PR)

**Comment by hbagdi** [comment]
Rebased. @bowei Another `/lgtm`?

I'll send a PR for moving it once this is in.

**Pull request merged**